feature_engine/timeseries/forecasting/time_lag.py

```diff
@@ -0,0 +1,173 @@
+# Authors: Morgan Sell <morganpsell@gmail.com>
+# License: BSD 3 clause
+
+from typing import List, Union
+
+import pandas as pd
+from pandas.tseries.offsets import DateOffset
+from pandas.tseries.frequencies import to_offset
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_is_fitted
+
+from datetime import datetime
+
+from feature_engine.dataframe_checks import (
+    _is_dataframe,
+)
+from feature_engine.variable_manipulation import (
+    _find_all_variables,
+    _check_input_parameter_variables,
+)
+
+
+class TimeSeriesLagTransformer(BaseEstimator, TransformerMixin):
+    """
+
+    Parameters
+    ----------
+    periods: int, default=1
+        Number of periods to shift. Can be positive or negative.
+
+    freq: str, default=None
+        Optional increment to use from the tseries module or time rule.
+
+    axis: int, default=1
+        Shift direction. Index is '0'. Columns are '1'.
+
+    fill_value: [int, float, str], default=None
+        Character to be used to fill missing values.
+
+    keep_original: bool, default=True
+        Determines whether the dataframe keeps the original columns
+        that are transformed.
+
+    Attributes
+    ----------
+
+
+    Methods
+    -------
+
+
+    Notes
+    -----
+
+
+    See Also
+    --------
+
+
+    References
+    ----------
+
+    """
+
+    def __init__(
+        self,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+        periods: int = 1,
+        freq: str = None,
+        axis: int = 0,
+        fill_value: [int, str] = None,
+        keep_original: bool = True,
```

Comment:
normally we call this param drop_original and we default to False. Can we change? so we are compatible with our other transformers?